### PR TITLE
LINK-93, LINK-94, LINK-95 | Event keywords

### PR DESCRIPTION
**Pull request by jorilindell**
## Description
Add GraphQl queries to get events
Generic CheckboxGroupField component
KeywordSelector component to search and select keywords
Audience and classification event form sections

## Closes
[LINK-93](https://helsinkisolutionoffice.atlassian.net/browse/LINK-93)
[LINK-94](https://helsinkisolutionoffice.atlassian.net/browse/LINK-94)
[LINK-95](https://helsinkisolutionoffice.atlassian.net/browse/LINK-95)

## Screenshots
Screenshot 2020-11-20 at 14 18 08

Screenshot 2020-11-20 at 14 18 16

### CODE REVIEW

**Comment by Sartonon** [comment]
Screenshot 2020-11-22 at 16 58 31

Seems like safari doesn't like the way form steps are handled in the dom, form gets too wide when navitagingto next step

**Comment by jorilindell** [comment]
> Screenshot 2020-11-22 at 16 58 31
> 
> Seems like safari doesn't like the way form steps are handled in the dom, form gets too wide when navitagingto next step

Fixed. Didn't happen with my 13 inch Macbook, but only with larger screens. Also fixed css issues with loading spinner and Safari

**Review by Sartonon — COMMENTED** [review]
hmm is there a reason not to use events-helsinki-api-proxy? Altough apollo-link-rest seems handy :) Otherwise looking good!

File: src/common/components/keywordSelector/KeywordSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  }, [apolloClient, locale, value]);
+
+  return (
+    <>
+      {value.map((keyword, index) => {
+        return (
+          // Make sure all selected keywords are fetched to cache
+          <KeywordQuery key={index} id={parseIdFromAtId(keyword) as string} />
+        );
+      })}
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/common/components/keywordSelector/KeywordSelector.tsx]: It's not. I think that I changed the logic and forgot to remove this

File: src/common/components/keywordSelector/KeywordSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+type KeywordQueryProps = {
+  id: string;
+};
+
+const KeywordQuery: React.FC<KeywordQueryProps> = ({ id }) => {
+  // hook to fetch keyword so KeywordSelector can get keyword from the cache
+  useKeywordQuery({
+    variables: {
+      id,
+      createPath: isTestEnv
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/common/components/keywordSelector/KeywordSelector.tsx]: Added getPathBuilder helper function

File: src/common/components/keywordSelector/KeywordSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  const [search, setSearch] = React.useState('');
+  const [options, setOptions] = React.useState<OptionType[]>([]);
+  const [selectedKeywords, setSelectedKeywords] = React.useState<OptionType[]>(
+    []
+  );
+
+  const { data: keywordsData } = useKeywordsQuery({
+    variables: {
+      freeText: search,
+      createPath: isTestEnv
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/common/components/keywordSelector/KeywordSelector.tsx]: Added getPathBuilder helper function

File: src/domain/event/formSections/audienceSection/AudienceSection.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import { EVENT_FIELDS } from '../../constants';
+import InputWrapper from '../InputWrapper';
+
+const TargetGroupSection = () => {
+  const { t } = useTranslation();
+  const locale = useLocale();
+
+  const { data } = useKeywordSetQuery({
+    variables: {
+      createPath: isTestEnv
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/event/formSections/audienceSection/AudienceSection.tsx]: Added getPathBuilder helper function

File: src/domain/event/formSections/classificationSection/ClassificationSection.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+import styles from '../../eventPage.module.scss';
+import InputWrapper from '../InputWrapper';
+
+const ClassificationSection = () => {
+  const { t } = useTranslation();
+  const locale = useLocale();
+
+  const { data } = useKeywordSetQuery({
+    variables: {
+      createPath: isTestEnv
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/event/formSections/classificationSection/ClassificationSection.tsx]: Added getPathBuilder helper function

File: src/domain/keyword/utils.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+export const getKeywordFromCache = async (
+  id: string,
+  apolloClient: ApolloClient<object>
+): Promise<Keyword | null> => {
+  try {
+    const { data: keywordData } = await apolloClient.query<KeywordQuery>({
+      query: KeywordDocument,
+      variables: {
+        id,
+        createPath: isTestEnv
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/keyword/utils.ts]: Added getPathBuilder helper function

**Comment by jorilindell** [comment]
> hmm is there a reason not to use events-helsinki-api-proxy? Altough apollo-link-rest seems handy :) Otherwise looking good!

So far events-helsinki-api-proxy is only for fetching data from the LinkedEvents. On next sprint we're going to add authentication and saving data features to the UI which events-helsinki-api-proxy doesn't support. At some there will be GraphQl API and we have to migrate to that anyway. I think that events-helsinki-api-proxy could start using that when it's available.

**Review by Sartonon — APPROVED** [review]

**Comment by Sartonon** [comment]
> > hmm is there a reason not to use events-helsinki-api-proxy? Altough apollo-link-rest seems handy :) Otherwise looking good!
> 
> So far events-helsinki-api-proxy is only for fetching data from the LinkedEvents. On next sprint we're going to add authentication and saving data features to the UI which events-helsinki-api-proxy doesn't support. At some there will be GraphQl API and we have to migrate to that anyway. I think that events-helsinki-api-proxy could start using that when it's available.

OK!

**Pull request merged**